torch/quantization/convert_modules.py

```diff
@@ -0,0 +1,222 @@
+    splits = name.split('.')
+    print('splits:', splits)
+    for split in splits:
+        parent_module = module
+        module = module._modules[split]
+    assert parent_module is not module
+    parent_module._modules[splits[-1]] = QuantWrapper(module, qconfig)
+
+def add_quant_dequant(module, qconfig_dict):
+    r"""Add QuantStub and DeQuantStub module and Wrap the module in QuantWrapper
+    according to qconfig_dict
+    """
+    mod_key_set = set()
+    for name, _ in module.named_modules():
+        print('name', name)
+        dict_key = get_config_key(name, qconfig_dict)
+        if dict_key is not None:
+            mod_key_set.add(dict_key)
+
+    # We need to update the modules from inner most
+    # module to outer modules(in decreasing order of number of '.')
+    print('mod_key_set', mod_key_set)
+    mod_key_heap = []
+    for name in mod_key_set:
+        heapq.heappush(mod_key_heap, (-name.count('.'), name))
+    while mod_key_heap:
+        _, name = heapq.heappop(mod_key_heap)
+        if name == '':
+            module = QuantWrapper(module, qconfig_dict[name])
+        else:
+            add_quant_dequant_module(module, qconfig_dict, name)
+    return module
+
+def prepare(module, qconfig_dict):
+    r"""Transform a module to a quantized module according to qconfig_dict,
+    it adds observer and quant dequant stub modules to the module. and
+    changes the forward method to invoke them
+
+    Args:
+        mod: instance of the module we want to transform
+        qconfig_dict: dictionary that maps from name of submodule to quantization
+                     configuration
+    """
+    add_observer(module, qconfig_dict)
+    return add_quant_dequant(module, qconfig_dict)
+
+class QuantStub(nn.Module):
+    r"""Quantize stub module which will be replace to actual Quantize module
+    in swap_module function
+    """
+    def __init__(self, qconfig):
+        super(QuantStub, self).__init__()
+        self.add_module('observer', qconfig.activation())
+
+    def forward(self, x):
+        self.observer(x)
+        return x
+
+class DeQuantStub(nn.Module):
+    r"""Dequantize stub module which will be replace to actual DeQuantize module
+    in swap_module function
+    """
+    def __init__(self):
+        super(DeQuantStub, self).__init__()
+
+    def forward(self, x):
+        return x
+
+def quantize(module, qconfig_dict, eval_fn, *eval_args):
+    r"""Given a float module and qconfig_dict, convert it to a quantized module
+
+    First it will prepare the module to add observer and quant
+    deqaunt stub modules, then it calls `eval_fn` which will run the calibration
+    step and observers will record the stats of the tensors, after that we will
+    call `convert` which will convert the module to a quantized module.
+    """
+    module = prepare(module, qconfig_dict)
+    eval_fn(module, *eval_args)
+    convert_to_quantized(module)
+    return module
+
+def convert_to_quantized(module):
```

Comment:
mapping should be the argument here as well (as that's the top api)

Reply:
not addressed

Reply:
By mapping you mean the dictionary of module -> quant modules ? I feel that should not be specified by the user: As we add support for more nn.quantized modules, shouldnt it get automatically updated?  User can anyway control which module gets quantized by qconfigdict.

Reply:
What if I define my own module though? We can provide the default map (which should be a default value), but it should be overridable imho

Reply:
@raghuramank100 should be fine, user can override default by DEFAULT + {...}